### PR TITLE
Strip trailing non-alphanumeric characters from importGroup when creating config.textproto

### PR DESCRIPTION
**Pull request by n-h-diaz**

### CODE REVIEW

File: gcf/custom/manifest.go [hunk 1]
```diff
@@ -25,6 +26,40 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+func isAlphaNumeric(c byte) bool {
+	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9')
+}
+
+func computeImportGroup(root string) (string, error) {
```
Comment (shifucun) [inline gcf/custom/manifest.go:33]: getImportGroupName?

File: gcf/custom/manifest.go [hunk 2]
```diff
@@ -25,6 +26,40 @@ import (
...
+
+	// Strip any trailing non-alphanumeric characters.
+	i := len(importGroup)
+	for !isAlphaNumeric(importGroup[i-1]) {
+		i--
+		if i == 0 {
+			break
+		}
+	}
+	importGroup = importGroup[:i]
```
Comment (shifucun) [inline gcf/custom/manifest.go:54]: why not do the trimming first then take the first 20?

File: gcf/custom/manifest.go (hunk 1 above)
Comment (n-h-diaz) [reply gcf/custom/manifest.go]: Done

File: gcf/custom/manifest.go (hunk 2 above)
Comment (n-h-diaz) [reply gcf/custom/manifest.go:54]: This is to handle cases where the truncated name ends with symbols 
(e.g. "sustainable_systems_lab" --> "sustainable_systems_" --> "sustainable_systems") 

Alternatively I could just remove all non-alphanumeric characters from the name

Wdyt?

Comment (shifucun) [reply gcf/custom/manifest.go:54]: Yea, that sg and leaves more room for truncation!

Comment (shifucun) [reply gcf/custom/manifest.go:54]: oh, do you want to make the change to remove non-alphanum chars?

Comment (n-h-diaz) [reply gcf/custom/manifest.go]: Done

**Review by shifucun — APPROVED** [review]
thanks!

**Pull request merged**